vowpalwabbit/learner.cc

```diff
@@ -186,15 +175,21 @@ class multi_example_handler
       _context.template process<multi_ex, learn_multi_ex>(ec_seq);
       ec_seq.clear();
     }
-
-    // Send out the end-of-pass notification after doing learning
+    // after we learn, cleanup is_newline or end_pass example
     if (ec->end_pass)
     {
       // Because the end_pass example is used to complete the in-flight multi_ex prior
       // to this call we should have no more in-flight multi_ex here.
       assert(ec_seq.empty());
       _context.template process<example, end_pass>(*ec);
     }
+    else if (ec->is_newline)
+    {
+      // Because the is_newline example is used to complete the in-flight multi_ex prior
+      // to this call we should have no more in-flight multi_ex here.
+      assert(ec_seq.empty());
```

Comment:
If there are two new lines in a row, will this get triggered?

Reply:
yes, if it's a newline that comes right after the multi-line examples then learn will be called and then finish on the first newline. Then the second newline will not trigger a learn and will just be cleaned up